src/pgwire/src/protocol.rs

```diff
@@ -356,9 +357,50 @@ where
             Some(Statement::Fetch(stmt)) => {
                 let name = stmt.name.clone();
                 let count = stmt.count;
+                let options = match FetchOptions::try_from(stmt.options.clone()) {
+                    Ok(options) => options,
+                    Err(e) => {
+                        return Some(
+                            self.error(ErrorResponse::error(
+                                SqlState::INVALID_PARAMETER_VALUE,
+                                format!("{}", e),
+                            ))
+                            .await,
+                        )
+                    }
+                };
+                let timeout_secs = match options.timeout {
+                    Some(timeout) => {
+                        // Limit FETCH timeouts to 1 day. If users have a legitimate need it can be
+                        // bumped. If we do bump it, ensure that the new upper limit is within the
+                        // bounds of a tokio::time::Duration.
```

Comment:
nit: it's not the bounds of `tokio::time::Duration` that are the issue; it's the bounds of what Tokio's time futures can support.